openedx/features/assessment/tests/test_helpers.py

```diff
@@ -0,0 +1,247 @@
+
+        self.enrollment = self._create_user_enrollments(self.user, THREE_WEEKS_AGO)
+
+        self.course_chapters = modulestore().get_items(
+            self.source_course.id,
+            qualifiers={'category': 'course'}
+        )
+
+    @mock.patch('openedx.features.assessment.helpers.log.info')
+    def test_log_multiple_submissions_info_with_multiple_submissions(self, mock_log_info):
+        SubmissionFactory.create()
+        SubmissionFactory.create()
+        submissions = Submission.objects.all()
+        helpers._log_multiple_submissions_info(submissions)
+        assert mock_log_info.called_once
+
+    @mock.patch('openedx.features.assessment.helpers.log.info')
+    def test_log_multiple_submissions_info_with_no_submissions(self, mock_log_info):
+        submissions = []
+        helpers._log_multiple_submissions_info(submissions)
+        assert mock_log_info.not_called
+
+    @mock.patch('openedx.features.assessment.helpers.log.info')
+    @mock.patch('openedx.features.assessment.helpers.reset_score')
+    @mock.patch('openedx.features.assessment.helpers.set_score')
+    def test_autoscore_ora(self, mock_log_info, mock_reset_score, mock_set_score):
+        """ Mock test score methods when there is valid submission exists """
+
+        chapter1 = self.course_chapters[0].children[0]
+        chapter1_sequential1 = modulestore().get_item(chapter1).children[0]
+        chapter1_sequential1_vertical1 = modulestore().get_item(chapter1_sequential1).children[0]
+        chapter1_sequential1_vertical1_block1 = modulestore().get_item(chapter1_sequential1_vertical1).children[0]
+
+        anonymous_user = AnonymousUserIdFactory.create(user=self.user, course_id=self.source_course.id)
+
+        # create submission item
+        student_item = StudentItemFactory.create(
+            student_id=anonymous_user.anonymous_user_id,
+            course_id=self.source_course.id,
+            item_id=chapter1_sequential1_vertical1_block1
+        )
+        submission = SubmissionFactory.create(
+            student_item=student_item,
+            created_at=FOUR_DAYS_AGO
+        )
+        student = {
+            'anonymous_user_id': anonymous_user.anonymous_user_id
+        }
+        helpers.autoscore_ora(self.source_course.id, unicode(chapter1_sequential1_vertical1_block1), student)
+
+        # Find the associated rubric for that course_id & item_id
+        rubric_dict = helpers.get_rubric_for_course(self.source_course.id, unicode(chapter1_sequential1_vertical1_block1))
+        options_selected, earned, possible = helpers.select_options(rubric_dict)
+
+        assert mock_log_info.called_once
+        assert mock_reset_score.called_with(anonymous_user.anonymous_user_id,
+                                            self.source_course.id.to_deprecated_string(), unicode(chapter1_sequential1_vertical1_block1))
+        assert mock_set_score.called_with(str(submission.uuid), earned, possible)
```

Comment:
Lets also assert entries in `Assessment` and `AssessmentPart` created successfully

Reply:
done